lib/delta-table/src/Demo/Database.hs

```diff
@@ -66,5 +70,22 @@ action = do
 
 main :: IO ()
 main = do
-    rows <- Sqlite.withConnection ":memory:" $ Sql.runSqlM action
+    rows <- Sql.withConnection ":memory:" $ Sql.runSqlM action
     for_ rows print
+
+testExceptions :: IO ()
+testExceptions = do
+    Sql.withConnection ":memory:" $ \conn -> do
+        _ <- Sql.runSqlM action conn
+        putStrLn "Before"
+        printTablePerson conn
+        Sql.runSqlM
+            ( do
+                Sql.deleteWhere (colBirthYear Sql.>. 1800) tablePerson
+                error "oops"
+            ) conn `catch` (\(_ :: SomeException) -> pure ())
+        putStrLn "After"
+        printTablePerson conn
+  where
+    printTablePerson conn =
+        mapM_ print =<< Sql.runSqlM (Sql.selectAll tablePerson) conn
```

Comment:
Did you consider writing these demos as simple unit tests from the beginning? 

I imagine it would be little additional work to e.g. replace `printTablePerson conn` with

```haskell
readTablePerson conn
    `shouldReturn`
    [ ("Babbage",1791)
    , ("William",1805)
    , ("Ada",1815)
    ]
```

Reply:
> Did you consider writing these demos as simple unit tests from the beginning?

Since I didn't know what the interface will look like, it didn't make sense for me to write the demo as a unit test in the beginning. But I agree that it makes much more sense now, and that's what I intend to do — in a future pull request. 😅